### PR TITLE
solution

### PR DESCRIPTION
**Pull request by olenitut**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,19 @@
...
+    } else if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach(key => {
+        delete state[key];
+      });
+    } else {
+      Object.keys(state).forEach(key => {
+        delete state[key];
+      });
+    }
+  });
```
Comment (Bohdan-Maksymiuk) [inline src/transformState.js:20]: avoid such nesting `else if` conditions
This code is hard to read and maintain

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]
try to use `switch case` for such predefined cases

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,22 @@
...
+      case 'removeProperties':
+        action.keysToRemove.forEach(key => {
+          delete state[key];
+        });
+        break;
+      default :
+        Object.keys(state).forEach(key => {
+          delete state[key];
+        });
+    }
```
Comment (Bohdan-Maksymiuk) [inline src/transformState.js:22]: believe that it should be default behaviour
what if someone will pass invalid data - state will be cleared by default
[usefulLink](https://www.quora.com/What-is-the-purpose-of-default-in-a-switch-case)
[usefulLink](https://softwareengineering.stackexchange.com/questions/181459/is-it-necessary-to-add-the-default-case-while-using-switch-cases)

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,22 @@
...
+      case 'removeProperties':
+        action.keysToRemove.forEach(key => {
+          delete state[key];
+        });
+        break;
+      case 'clear':
+        Object.keys(state).forEach(key => {
+          delete state[key];
+        });
+    }
```
Comment (Viktor-Kost) [inline src/transformState.js:22]: You would better use `default` for switch statement. Cause what will happen if we pass a case we don't have?
For example for default you can throw an error `throw new Error('invalid type of action')`

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]
One more little stuff.
Have given a recommendation

**Review by Viktor-Kost — APPROVED** [review]
Well done!

**Pull request closed without merging**